op-plasma/dastate.go

```diff
@@ -146,6 +170,10 @@ func (s *State) ExpireChallenges(bn uint64) (uint64, error) {
 	var err error
 	for i := 0; i < len(s.comms); i++ {
```

Comment:
hmm, I think I missed this usage of `s.comms`. It's broken & explains why we ran into the stall when I challenged the first commitment. `s.comms` is a PQ so it's the wrong data structure for the job. If we know that challenges always expire `n` L1 blocks are they are initiated, then we can use `s.comms` as a fifo queue with the knowledge that `s.comms` is order by `c.expiresAt` (because we process L1 blocks in order).

If the relation is a bit looser, we need to find an alternate datastructure like a b-tree map.